core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4038,7 +4066,7 @@ public static Collection multisetUnionAll(Collection collection1,
   }
 
   /** Support the ARRAY_REVERSE function. */
-  public static List reverse(List list) {
+  public static List arrayReverse(List list) {
```

Comment:
why are we going to rename public methods?
that could impact downstream dependencies

Reply:
1.35 has not been released yet, it will not affect downstream dependencies.

The old name is not accurate enough, we should solve this problem before the release version.